### PR TITLE
Userland elf loading

### PR DESCRIPTION
**Pull request by Skallwar**
Far from finished but I like CI

### CODE REVIEW

File: kernel/src/arch/riscv64/sv39.rs [hunk 1]
```diff
@@ -159,8 +190,13 @@ pub struct PageTable {
 }
 
 impl PageTable {
+    fn base(&self) -> usize {
+        self.entries.as_ptr() as usize
+    }
+
     fn map_inner(
         &mut self,
+        mut kernel_page_table: Option<&mut Self>,
```
Comment (n1tram1) [inline kernel/src/arch/riscv64/sv39.rs:199]: oskour

**Comment by Skallwar** [comment]
I think the allocator should have an `alloc_mapped()` method to avoid this pattern alloc + mapping everywhere. Plus

**Comment by Skallwar** [comment]
At this stage we have a working userland program running on RISC-V. I managed to run it as a test.

I think the main ideas and the implementation is decent but I guess the form needs brilliant ideas form you guys.

@n1tram1 AArch64 support? :eyes:

**Comment by Skallwar** [comment]
This is almost ready. Userland for Aarch64 is underway, but I forgot that we needed a pagetable for Aarrch64 first

**Comment by Skallwar** [comment]
Rebased on top of #115

**Comment by Skallwar** [comment]
Closed in favor of #132

**Pull request closed without merging**